### ISSUE TITLE
Breakpoints are ignored in packages

### ISSUE THREAD

**Issue by faustinoaq** [DonJayamanne/javaVSCode#16]
Hi @DonJayamanne ! :smile: 
- VSCode Version: 1.6.1
- OS Version: Manjaro KDE Ellada 16.08

Steps to Reproduce: _(when breakpoints are ignored)_
1. Make a project with `package myawesomepackage`
2. Install `Java Debugger` in VSCode
3. Configure `launch.json` and `tasks.json`
4. Create `breakpoints`
5. Execute the debugger

Only `Java Debugger` exist for java debugging, and works very well. :+1: 

When I try to debug `netbeans | eclipse | mvn` projects I just setup my `launch.json` and `tasks.json` in `.vscode`. But the brekpoints are ignored when a package is used for the project.
### With package

I can debug the code _by step_ but the breakpoints are ignored.

![java-with-package](http://i.imgur.com/4yKD4lu.gif)
### Without package

I fixed temporally removing the package line, then setup new classpath directory in `launch.json` and `tasks.json`

![java-without-package](http://i.imgur.com/rrNwjqP.gif)

Maybe related? #1 #11 but my issue is when java package exist.

Bye! :raising_hand_man:

**Comment by DanJ210** [comment]
@faustinoaq Ah yes! I am having all the same experiences! I like how issues can be referenced! The above is the same for me.

**Comment by sunnyliujing** [comment]
Hi @faustinoaq ,how do you configure the build task?And this is my configuration:
`tasks.json` 

```
{
    "version": "0.1.0",
    "command": "javac",
    "isShellCommand": true,
    "showOutput": "always",
    "isWatching": true,
    "suppressTaskName": true,
    "jdkPath": "/Library/Java/JavaVirtualMachines/jdk1.8.0_101.jdk/Contents/Home/bin",
    "tasks": [
        {
            "taskName": "build",
            "args": ["-g", "${file}"]
        }
    ]
}
```

`launch.json`

```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Java",
      "type": "java",
      "request": "launch",
      "stopOnEntry": true,
      "jdkPath": "/Library/Java/JavaVirtualMachines/jdk1.8.0_101.jdk/Contents/Home/bin",
      "cwd": "${workspaceRoot}",
      "startupClass": "${fileBasename}",
      "options": [
        "-classpath",
        "${workspaceRoot}"
      ]
    }
  ]
}
```

I don't have the package.the problem is could not find or load main class XXX(the java file name)

**Comment by faustinoaq** [comment]
@sunnyliujing see [my comment](https://github.com/DonJayamanne/javaVSCode/issues/19#issuecomment-259208604)

`task.json` :point_down: 
```json
{
  "version": "0.1.0",
  "command": "javac",
  "isShellCommand": true,
  "args": [
    "-g",
    "-cp",
    "${workspaceRoot}/bin",
    "-d",
    "${workspaceRoot}/bin",
    "${workspaceRoot}/src/TestingJava.java"
  ],
  "showOutput": "always"
}
```

**Comment by ThePhilderbeast** [comment]
ok I had a little look at this, but I'm by no means a JS/TS dev so I'm not sure how to fix it.

it looks like when registering the breakpoints with JDB its only giving it the class name and not the full qualified name with the package name. 

I think its in main.ts around line 282 where it is trunkating the name to the base class name, but i would need to do more reading to work out what that needs changing to

**Comment by viniciuscr** [comment]
Same problem here.
When i start debug it stops at the very first code line, i can walk trough all code by steps using F10 and F11, but i can´t use resume because the breakpoints are ignored.

**Comment by tcsx** [comment]
Setting the "sourcepath" in launch.json fixes it for me. See #23 , it has fixed #22 .

**Comment by faustinoaq** [comment]
Hi @tcsx Thanks you for the suggestion but setting the `sourcepath` dont work for me.

Can you give a repo example where you use package and breakpoints works?

this is my actual `launch.json`

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Java Console App",
      "type": "java",
      "request": "launch",
      "stopOnEntry": true,
      "jdkPath": "/usr/lib/jvm/default/bin",
      "cwd": "${workspaceRoot}",
      "startupClass": "my.greet.Hello",
      "sourcePath": [
        "${workspaceRoot}/src/"
      ],
      "options": [
        "-classpath",
        "${workspaceRoot}/bin"
      ]
    }
  ]
}
```

**Comment by faustinoaq** [comment]
I found the problem. Is like @ThePhilderbeast said.
Breakpoints don't work in packages because `setBreakPointsRequest` generate className from path and ignore if the program is inside a package.

You can see the code here : [L282](https://github.com/DonJayamanne/javaVSCode/blob/master/src/client/main.ts#L282)

If I hardcode my className inside extension then breakpoints works well.

![screenshot_20170609_172131](https://user-images.githubusercontent.com/3067335/26996485-2180401e-4d38-11e7-8b3f-538d9eed7708.png)

**Comment by faustinoaq** [comment]
I fixed my issue using ```className = `${className.toLowerCase()}.${className}`;``` to follow Netbeans Project pattern where by default package name is the lowercase of the main class.

![screenshot_20170609_181309](https://user-images.githubusercontent.com/3067335/26997440-4f48792e-4d3f-11e7-979a-93c122b4edbb.png)



Perhaps a solutions is #25 However is hard to determine package because depend of project structure: maven, eclipse, netbeans, intellij, etc.

**Comment by ThePhilderbeast** [comment]
this isn't fixed so probably shouldn't be closed.

**Comment by faustinoaq** [comment]
Yeah, I hacked this extension a bit, i 'll keep this issue open.

Edit: Fixed with PR #42
